### PR TITLE
Pro issue 1460

### PR DESCRIPTION
**Pull request by BeckmaR**
This pull request allows to check imported resources for errors and display them on the import statement.

A good way to sync errors from imported resources with the definition section is missing; the only possible way seems to be to revalidate the definition section when new issues are found on external file changes. This leads to the validation being done twice.

### CODE REVIEW

**Comment by andreasmuelder** [comment]
Do not merge for 3.4.2

File: plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java [hunk 1]
```diff
@@ -899,34 +905,51 @@ public void checkImportExists(ImportScope scope) {
 				error(String.format(IMPORT_NOT_RESOLVED_MSG, packageImport), scope,
 						StextPackage.Literals.IMPORT_SCOPE__IMPORTS, imports.indexOf(packageImport));
 			}
+			else {
+				IResourceDescription iResourceDescription = resourceDescriptionCache.get(pkImport.get().getUri());
+				if (iResourceDescription instanceof DefaultResourceDescription) {
+					Resource resource = ((DefaultResourceDescription) iResourceDescription).getResource();
+					if (!(resource.getWarnings().isEmpty() && resource.getErrors().isEmpty())) {
```
Comment (tkutz) [inline plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextJavaValidator.java]: Does it mean, I will now get an error in the statechart although the imported resource has just a warning?
I would prefer to not be too strict here. Actually I think an info message would be enough here.

**Comment by BeckmaR** [comment]
@tkutz changes implemented

**Comment by andreasmuelder** [comment]
closing, this will not be part of the 3.x maintainance

**Pull request closed without merging**